README.md

```diff
@@ -30,7 +33,107 @@ sending out wanted blocks if they have them. Upon receiving blocks, the node
+
+## Usage
+
+### Initializing a Bitswap Exchange
+
+```golang
+import (
+  "context"
+  bitswap "github.com/ipfs/go-bitswap"
+  bsnet "github.com/ipfs/go-graphsync/network"
+  blockstore "github.com/ipfs/go-ipfs-blockstore"
+  "github.com/libp2p/go-libp2p-core/routing"
+	"github.com/libp2p/go-libp2p-core/host"
+)
+
+var ctx context.Context
+var host host.Host
+var router routing.ContentRouting
+var bstore blockstore.Blockstore
+
+network := bsnet.NewFromIPFSHost(host, router)
+exchange := bitswap.New(ctx, network, bstore)
+```
+
+Parameter Notes:
+
+1. `ctx` is just the parent context for all of Bitswap
+2. `network` is a network abstraction provided to Bitswap on top
+of libp2p & content routing. 
+3. `bstore` is an IPFS blockstore
+
+### Get A Block Synchronously
+
+```golang
+var c cid.Cid
+var ctx context.Context
+var exchange bitswap.Bitswap
+
+block, err := exchange.GetBlock(ctx, c)
+```
+
+Parameter Notes:
+
+1. `ctx` is the context for this request, which can be cancelled to cancel the request
+2. `c` is the content ID of the block you're requesting
+
+### Get Several Blocks Asynchronously
+
+```golang
+var cids []cid.Cid
+var ctx context.Context
+var exchange bitswap.Bitswap
+
+blockChannel, err := exchange.GetBlocks(ctx, cids)
+```
+
+Parameter Notes:
+
+1. `ctx` is the context for this request, which can be cancelled to cancel the request
+2. `cids` is an slice of content IDs for the blocks you're requesting
+
+### Get Related Blocks Faster With Sessions
+
+In IPFS, content blocks are often connected to each other through a MerkleDAG. If you know ahead of time that block requests are related, Bitswap can make several optimizations internally in how it requests those blocks in order to get them faster. To facilitate this, Bitswap provides a mechanism called a Bitswap session to designate a series of block requests being part of a single higher level operation. You should initialize a bitswap session any time you intend to make a series of block requests that related -- and whose responses are likely to come from the same peers. 
+
+```golang
+var ctx context.Context
+var cids []cids.cid
+var exchange bitswap.Bitswap
+
+session := exchange.NewSession(ctx)
+blocksChannel, err := session.GetBlocks(ctx, cids)
+// later
+var relatedCids []cids.cid
+relatedBlocksChannel, err := session.GetBlocks(ctx, relatedCids)
+```
+
+Note that new session returns an interface with a GetBlock and GetBlocks method that have the same signature as the overall Bitswap exchange.
+
+### Tell bitswap a new block was added to the local datastore
+
+```golang
+var blk blocks.Block
+var exchange bitswap.Bitswap
+
+err := exchange.HasBlock(blk)
+```
+
 ## Implementation
+
+The following diagram outlines the major tasks Bitswap handles, and their consituent components:
+
+![Bitswap Components](./docs/go-bitswap.png)
+
+### Sending Blocks
+
 Internally, when a message with a wantlist is received, it is sent to the
```

Comment:
Again not part of this PR, but I'd suggest changing this paragraph to avoid using the passive voice